datasets/imagenet/imagenet.py

```diff
@@ -0,0 +1,195 @@
+"""
+
+_VAL_PREP_SCRIPT_IN1K = "https://raw.githubusercontent.com/soumith/imagenetloader.torch/master/valprep.sh"
+_SYNSET_MAPPING_IN1K = "https://gist.githubusercontent.com/apsdehal/f27ebf7d594b5950c91ecf732c0aaf61/raw/0b4cd44dadf7385e6d656424911cf45a113685bc/imagenet_synsets.txt"  # noqa
+_DEFAULT_CONFIG_NAME = "2012"
+
+
+class ImagenetConfig(datasets.BuilderConfig):
+    def __init__(
+        self,
+        name=_DEFAULT_CONFIG_NAME,
+        val_prep_script=_VAL_PREP_SCRIPT_IN1K,
+        synset_mapping=_SYNSET_MAPPING_IN1K,
+        **kwargs,
+    ):
+        """Config for ImageNet dataset.
+
+        Args:
+            name (string, optional): keyword to identify the config. Defaults to _DEFAULT_CONFIG_NAME.
+            val_prep_script (string, optional): URL or path to script used to prep val set. Defaults to _VAL_PREP_SCRIPT_IN1K.
+            synset_mapping (string, optional): URL or path to synset mapping. Defaults to _SYNSET_MAPPING_IN1K.
+        """
+        kwargs.pop("version", None)
+        super(ImagenetConfig, self).__init__(version=datasets.Version("1.0.0"), name=name, **kwargs)
+        self.val_prep_script = val_prep_script
+        self.synset_mapping = synset_mapping
+
+
+class Imagenet(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIG_CLASS = ImagenetConfig
+
+    BUILDER_CONFIGS = [
+        ImagenetConfig(
+            name="2012",
+            val_prep_script=_VAL_PREP_SCRIPT_IN1K,
+            synset_mapping=_SYNSET_MAPPING_IN1K,
+        ),
+    ]
+
+    IMAGE_EXTENSION = ".jpeg"
+
+    @property
+    def manual_download_instructions(self):
+        return (
+            "To use ImageNet you have to download it manually. Please download it from ("
+            "https://www.kaggle.com/competitions/imagenet-object-localization-challenge/"
+            "data?select=imagenet_object_localization_patched2019.tar.gz). You will "
+            "need to login and download the `imagenet_object_localization_patched2019.tar.gz file. "
+            "Don't extract the file and point the `datasets` library to the tar file by running "
+            "`datasets.load_dataset('imagenet', "
+            "data_dir='path/to/folder/imagenet_object_localization_patched2019.tar.gz')`"
+        )
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "image": datasets.Image(),
+                }
```

Comment:
Could you also add the `labels` field with a `ClassLabel` feature type here please ?

Reply:
Each image from the dataset is mapped to a single label so maybe `label` is a better name than `labels`. Also, let's add `ImageClassification(image_column="image", label_column="label")` as a task template to the dataset info.

Reply:
The reason I didn't add labels is similar to why `ImageFolder` doesn't have labels initially but are added later as we don't know them ahead of the time. I will have to load the labels file before hand if we want to achieve this. Let me know what you think.

Reply:
I have adjusted the code to reflect the review comments.